src/stats.jl

```diff
@@ -208,29 +208,29 @@ function ess(chn::AbstractChains;
 		end
 	end
 
-    # Misc
-    lags = collect(0:250)
+    # Misc allocations.
     m = n_chain_orig * 2
     n = min(length.(parameter_vec[1])...)
+    maxlag = min(maxlag, n-1)
+    lags = collect(0:maxlag)
 
-    # Preallocate
+    # Preallocate B, W, varhat, and Rhat vectors for each param.
     B = Vector(undef, length(param))
     W = Vector(undef, length(param))
     varhat = Vector(undef, length(param))
+    Rhat = Vector(undef, length(param))
 
+    # calculate B, W, varhat, and Rhat for each param.
     for i in 1:length(param)
 		draw = parameter_vec[i]
 		p = param[i]
         allchain = mean(vcat([d for d in draw]...))
-        eachchain = Vector(undef, m)
-        s = Vector(undef, m)
-        for j in 1:m
-            eachchain[j] = mean(draw[j])
-            s[j] = (1/(n-1)) * sum((draw[j] .- eachchain[j]).^2)
-        end
+        eachchain = [mean(draw[j]) for j in 1:m]
+        s = [sum((draw[j] .- eachchain[j]).^2) / (n-1) for j in 1:m]
         B[i] = (n / (m - 1)) * sum((eachchain .- allchain).^2)
-        W[i] = (1/m) * sum(s.^2)
-        varhat[i] = (n-1)/n * W[i] + (1/n) * B[i]
+        W[i] = sum(s) / m
+        varhat[i] = (n-1)/n * W[i] + (1/n) * (B[i] / n)
```

Comment:
I think one of the division of the between-chain variance by the chain length is superfluous here

Reply:
Ah, good catch. Thanks!